### PR TITLE
Incorporated Geolocation and restyled nav bar

### PR DESCRIPTION
**Pull request by jdmchugh111**
I used a javascript file to run a script on the welcome page when you click the "Use My Current Location" button. When that script runs it uses the Geolocation API to ask a user to share their location, pulls in their coordinates, then uses a geocoder api (opencagedata.com) to reverse geocode the latitude and longitude into a city and then populate that value into the search field.

I also restyled the nav bar while I was at it - gonna send a screen record to the slack to get feedback

### CODE REVIEW

File: app/assets/javascripts/geolocation.js.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+document.addEventListener('DOMContentLoaded', function() {
```
Comment (zach-bergman) [inline app/assets/javascripts/geolocation.js.erb:1]: Thanks for figuring this all out! Looks kind of complicated, but reading through this I can mostly follow what is happening

File: app/controllers/geocoder_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class GeocoderController < ApplicationController
+
+  def city_from_location
```
Comment (zach-bergman) [inline app/controllers/geocoder_controller.rb:3]: again thank you for doing all of this

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,10 +1,15 @@
 <!DOCTYPE html>
-<% if current_user.present?  %>
-  <%= link_to 'Log Out', '/signout' %>
-  <%= current_user.name %>
-<% else %>
-  <%= link_to 'Log In', '/auth/google_oauth2' %>
-<% end %>
+<div class="header-container d-flex justify-content-end align-items-center bg-danger text-white p-3">
```
Comment (zach-bergman) [inline app/views/layouts/application.html.erb:2]: this looks great!

**Review by zach-bergman — APPROVED** [review]

File: app/assets/javascripts/geolocation.js.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  document.getElementById('useLocation').addEventListener('click', function() {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(function(position) {
+        const latitude = position.coords.latitude;
+        const longitude = position.coords.longitude;
+
+        const url = `/geocoder/city_from_location?latitude=${latitude}&longitude=${longitude}`;
```
Comment (RodrigoACG) [inline app/assets/javascripts/geolocation.js.erb:8]: This looks complicated but this is still amazing how you got all this to work.  Great job !!

**Review by RodrigoACG — APPROVED** [review]

**Pull request merged**